src/lib/sb-file-uploader-hoc.jsx

```diff
@@ -0,0 +1,241 @@
+import bindAll from 'lodash.bindall';
```

Comment:
I tried to keep as much as possible in this file the same as before. There are very few differences, aside from what was necessary to refactor it into a top-level HOC wrapping GUI.

Reply:
> I tried to keep as much as possible in this file the same as before. There are very few differences, aside from what was necessary to refactor it into a top-level HOC wrapping GUI.

Now there are more differences -- as part of making the <input> component dynamic, I reordered functions and added comments.